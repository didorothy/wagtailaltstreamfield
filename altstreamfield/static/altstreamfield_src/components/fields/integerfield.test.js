import React from "react";
import { render, fireEvent, wait, waitForDomChange, waitForElementToBeRemoved, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import IntegerField from "./integerfield";
import FormErrorContext from "../../context/formerror";


describe('IntegerField', () => {
    test('#minimal', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={1} name="test" onChange={() => {}} />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#label', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={1} name="test" onChange={() => {}} label="Test Char Field" />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#help_text', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={1} name="test" onChange={() => {}} label="Test Char Field" help_text="Sample help text." />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#required', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={1} name="test" onChange={() => {}} label="Test Char Field" help_text="Sample help text." required={true} />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#error', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={null} name="test" onChange={() => {}} label="Test Char Field" help_text="Sample help text." required={true} />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('min_value', () => {
        let current_value = null;
        let onChange = (name, value) => {
            current_value = value;
        }
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} min_value={3}/>
        </FormErrorContext.Provider>);
        expect(manager.isValid()).toEqual(false);
        let input = container.container.getElementsByTagName('input')[0];
        let error = container.container.getElementsByClassName('error-message')[0];
        expect(error.textContent).toEqual('This field is required.')

        fireEvent.change(input, {target: {value: '1'}});
        container.rerender(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} min_value={3}/>
        </FormErrorContext.Provider>);
        expect(current_value).toEqual(1);
        expect(manager.isValid()).toEqual(false);

        fireEvent.change(input, {target: {value: '3'}});
        container.rerender(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} min_value={3}/>
        </FormErrorContext.Provider>);
        expect(current_value).toEqual(3);
        expect(manager.isValid()).toEqual(true);
        cleanup();
    });

    test('max_length', () => {
        let current_value = null;
        let onChange = (name, value) => {
            current_value = value;
        }
        let manager = new FormErrorContext.FormErrorManager();
        let container = render(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} max_value={3}/>
        </FormErrorContext.Provider>);
        expect(manager.isValid()).toEqual(false);
        let input = container.container.getElementsByTagName('input')[0];
        let error = container.container.getElementsByClassName('error-message')[0];
        expect(error.textContent).toEqual('This field is required.')

        fireEvent.change(input, {target: {value: "1"}});
        container.rerender(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} max_value={3}/>
        </FormErrorContext.Provider>);
        expect(current_value).toEqual(1);
        expect(manager.isValid()).toEqual(true);

        fireEvent.change(input, {target: {value: "4"}});
        container.rerender(<FormErrorContext.Provider value={manager}>
            <IntegerField owner_id="struct-01" value={current_value} name="test" onChange={onChange} label="Test Char Field" help_text="Sample help text." required={true} max_value={3}/>
        </FormErrorContext.Provider>);
        expect(current_value).toEqual(4);
        expect(manager.isValid()).toEqual(false);
        cleanup();
    });
});