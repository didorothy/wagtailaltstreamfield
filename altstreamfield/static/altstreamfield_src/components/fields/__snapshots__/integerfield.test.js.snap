// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntegerField #error 1`] = `
<div>
  <div
    class="field integer_field required error"
  >
    <label
      for="struct-01test"
    >
      Test Char Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="input"
      >
        <input
          id="struct-01test"
          placeholder="Test Char Field"
          type="number"
          value=""
        />
      </div>
      <p
        class="help"
      >
        Sample help text.
      </p>
      <p
        class="error-message"
      >
        This field is required.
      </p>
    </div>
  </div>
</div>
`;

exports[`IntegerField #help_text 1`] = `
<div>
  <div
    class="field integer_field"
  >
    <label
      for="struct-01test"
    >
      Test Char Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="input"
      >
        <input
          id="struct-01test"
          placeholder="Test Char Field"
          type="number"
          value="1"
        />
      </div>
      <p
        class="help"
      >
        Sample help text.
      </p>
      
    </div>
  </div>
</div>
`;

exports[`IntegerField #label 1`] = `
<div>
  <div
    class="field integer_field"
  >
    <label
      for="struct-01test"
    >
      Test Char Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="input"
      >
        <input
          id="struct-01test"
          placeholder="Test Char Field"
          type="number"
          value="1"
        />
      </div>
      
    </div>
  </div>
</div>
`;

exports[`IntegerField #minimal 1`] = `
<div>
  <div
    class="field integer_field"
  >
    <label
      for="struct-01test"
    >
      Test
    </label>
    <div
      class="field-content"
    >
      <div
        class="input"
      >
        <input
          id="struct-01test"
          placeholder="Test"
          type="number"
          value="1"
        />
      </div>
      
    </div>
  </div>
</div>
`;

exports[`IntegerField #required 1`] = `
<div>
  <div
    class="field integer_field required"
  >
    <label
      for="struct-01test"
    >
      Test Char Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="input"
      >
        <input
          id="struct-01test"
          placeholder="Test Char Field"
          type="number"
          value="1"
        />
      </div>
      <p
        class="help"
      >
        Sample help text.
      </p>
      
    </div>
  </div>
</div>
`;
